File: leetcode/python/43. Multiply Strings.py
class Main:

  @staticmethod
  def multiply(num1,num2):
    n1 = num1[::-1]
    n2 = num2[::-1]
    d = [0 for i in range(len(n1)+len(n2))]
    for i in range(len(n1)):
      for j in range(len(n2)):
        d[i+j] += int(n1[i]) * int(n2[j])
    result = ""
    for i in range(len(d)):
      digit = d[i] %  10
      carry = d[i] // 10
      if i+1 < len(d):
        d[i+1] += carry
      result = str(digit) + result
    index = 0
    while index < len(result) - 1 and result[index] == '0':
      index += 1
    return result[index:]


print(Main.multiply("7","8"))

File: leetcode/js/43. Multiply Strings.js
class Main {
  static multiply(num1, num2) {
    let n1 = num1.split("").reverse().join("");
    let n2 = num2.split("").reverse().join("");
    var d = Array(n1.length + n2.length).fill(0);
    for (let i = 0; i < n1.length; i++) {
      for (let j = 0; j < n2.length; j++) {
        d[i + j] += (n1[i] - "0") * (n2[j] - "0");
      }
    }
    let result = "";
    for (let i = 0; i < d.length; i++) {
      let digit = d[i] % 10;
      let carry = parseInt(d[i] / 10);
      if (i + 1 < d.length) {
        d[i + 1] += carry;
      }
      result = digit + "" + result;
    }
    let index = 0;
    while (index < result.length - 1 && result[index] == "0") {
      index++;
    }
    return result.substr(index);
  }
}

console.log(Main.multiply("7", "8"));
